client/src/main/java/com/cloudera/nav/sdk/client/writer/registry/RelationFactory.java

```diff
@@ -106,16 +113,32 @@ private Relation createInstanceOfRelation(RelationRole roleOfOther,
   private Relation createLogicalPhysicalRelation(
       RelationRole roleOfOther, Entity entity,
       Collection<? extends Entity> other, String namespace) {
+
+    Collection<IdAttrs> idAttrsList = getIdAttrsList(other);
     LogicalPhysicalRelation.Builder builder = LogicalPhysicalRelation.builder();
     if (roleOfOther == RelationRole.LOGICAL) {
       Preconditions.checkArgument(other.size() == 1,
           "Only 1 logical allowed in each logical-physical relationship");
-      builder.physical(entity).logical(Iterables.getOnlyElement(other));
+      builder.physical(entity).logical(Iterables.getOnlyElement(other))
+          .ep1Attributes(idAttrsList);
     } else {
-      builder.logical(entity).physical(other);
+      builder.logical(entity).physical(other).ep2Attributes(idAttrsList);
     }
     return builder.idGenerator(new RelationIdGenerator())
         .namespace(namespace).build();
   }
 
+  private Collection<IdAttrs> getIdAttrsList(
+      Collection<? extends Entity> other) {
+    Collection<IdAttrs> idAttrsList = Lists.newArrayList();
+    for (Entity en : other) {
+      if (Strings.isNullOrEmpty(en.getIdentity())) {
+        IdAttrs at = new IdAttrs();
+        en.populateIdAttrs(at);
```

Comment:
Why not just make a static method so you can combine the constructor call and the population into a single call?